chain/transaction_test.go

```diff
@@ -73,6 +75,58 @@ func unmarshalAction2(p *codec.Packer) (chain.Action, error) {
 	return &action, err
 }
 
+func TestJSONMarshalUnmarshal(t *testing.T) {
+	require := require.New(t)
+
+	txID := ids.GenerateTestID()
+	pk, err := ed25519.GeneratePrivateKey()
+	require.NoError(err)
+
+	tx := &chain.Transaction{
+		TransactionData: chain.TransactionData{
+			Actions: []chain.Action{
+				&mockTransferAction{
+					To:    codec.Address{1, 2, 3, 4},
+					Value: 4,
+					Memo:  []byte("hello"),
+				},
+				&mockTransferAction{
+					To:    codec.Address{4, 5, 6, 7},
+					Value: 123,
+					Memo:  []byte("world"),
+				},
+				&action2{
+					A: 2,
+					B: 4,
+				},
+			},
+		},
+		Auth: &auth.ED25519{
+			Signer: pk.PublicKey(),
+		},
+	}
```

Comment:
Could we use `NewTxData` and `Sign` to produce the signed transaction? This makes sure all fields are populated (currently missing the signature in the ED25519 auth)